### PR TITLE
Add tutorial for creating a basic prototype (based on GOV.UK one)

### LINKED ISSUE #17: Add a tutorial showing how to create a basic service prototype

### PR DESCRIPTION
**Pull request by vickytnz**
## Description
This is probably not quite ready to go straight to deployment because it has:

1. example pages to help with debugging or screenshots - these probably need to be deleted
2. a nunjucks base page with variables to help with change pages before it is resolved
3. code formatting that could be tidier
4. a few little elements of the nhs components that aren't yet in the code (for example, deleting the paragraph at the bottom of each template page that links to the templates)

It does have a full worked example bringing together the content into a guide.

Example of translated screenshots that are used in the examples

overview of new screens
new magical powers screen
new details screen

It's also shown that some of the design system components can do with tweaking, for example:
1. radio examples don't have the same text in the text as the variable 
2. radio example in component library starts with yes/no, unlike govuk design system which starts with list of 3
3. textarea component doesn't have headers shown

## Checklist

- [ ] Tested against our [testing policy](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/testing.md) (Resolution, Browser & Accessibility)
- [ ] CHANGELOG entry

Resolves #17

### CODE REVIEW

File: app/views/how-tos/make-first-prototype/open-prototype-in-editor.html
Comment (frankieroberto) [inline app/views/how-tos/make-first-prototype/open-prototype-in-editor.html:10]: Feels odd to go straight from an h1 to an h2. Copied across the content from the GOV.UK tutorial as an intro paragraph.

I also wonder whether an abstract diagram showing the layout of most editors (ie pane on the left with a folder view, and the right hand side having the editor with tabs) would help orientate people?

```suggestion
<p>In your code editor open your prototype folder. You will see the files and folders that you can edit in your prototype.</p>
```

File: app/views/how-tos/make-first-prototype/open-prototype-in-editor.html
Comment (frankieroberto) [inline app/views/how-tos/make-first-prototype/open-prototype-in-editor.html:13]: This first bullet is the lead-in I think?

```suggestion
<p><code>/app</code> is for your work. Inside that folder:</li></p>

<ul>
```

File: app/views/how-tos/make-first-prototype/create-pages.html
Comment (frankieroberto) [inline app/views/how-tos/make-first-prototype/create-pages.html:1]: There's a lot happening on this page (as on the GOV.UK one) - I wonder if it'd be better to go a bit more slowly and split across multiple pages, with example screenshots for each page?

Possibly one for a later iteration though?

File: app/views/how-tos/make-first-prototype/let-user-change-answers.html
Comment (frankieroberto) [inline app/views/how-tos/make-first-prototype/let-user-change-answers.html:24]: Doh, I didn’t know we even had this filter in the NHS prototype kit... I've been using `checked: data['magical-powers'] === 'yes'`.

At some point we should also add copy GOV.UK and let you set `value:` on the top-level macro option, to save having to set `checked` on each individual item, as that’d be far simpler.

File: app/views/how-tos/make-first-prototype/link-index-page-start-page.html
Comment (frankieroberto) [inline app/views/how-tos/make-first-prototype/link-index-page-start-page.html:15]: Fix link:

```suggestion
<p>You can now find out <a href="/how-tos/publish-your-prototype-online">how to publish your prototype online</a>, so you can share it with your team or do user research.</p>
```

File: app/views/ineligible.html
Comment (frankieroberto) [inline app/views/ineligible.html:1]: Presumably these pages should be deleted before we merge, and just used to test it out?

File: app/config.js
Comment (frankieroberto) [inline app/config.js:5]: Restore indenting here.

```suggestion
  serviceName: 'NHS.UK prototype kit documentation',
```

**Review by frankieroberto — CHANGES_REQUESTED** [review]
A great start! 

I think we’ll need to add some css for code blocks and inline code examples as those are currently quite hard to spot (just the font changes).

File: app/views/ineligible.html
Comment (vickytnz) [reply app/views/ineligible.html:1]: yep, exactly, it's particularly since we're using code examples and also doing screenshots - this makes it easier to debug (I also realise i haven't tested the branching)

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:24]: The rest of the lists on this page don’t use bullets, so match that? Also wrong prefix 😉 

```suggestion
      <ul class="nhsuk-list">
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:26]:
```suggestion
          <a href="/how-tos/make-first-prototype/start">Build a basic prototype</a>
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:27]:
```suggestion
          <ol class="nhsuk-list nhsuk-list--number">
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:29]:
```suggestion
              <a href="/how-tos/make-first-prototype/open-prototype-in-editor">Open your prototype in your editor</a>
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:32]:
```suggestion
              <a href="/how-tos/make-first-prototype/create-pages">Create pages</a>
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:35]:
```suggestion
              <a href="/how-tos/make-first-prototype/link-pages-together">Link your pages together</a>
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:38]:
```suggestion
              <a href="/how-tos/make-first-prototype/use-components">Use components from the Design System</a>
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:41]:
```suggestion
              <a href="/how-tos/make-first-prototype/show-users-answers">Show the user's answers</a>
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:44]:
```suggestion
              <a href="/how-tos/make-first-prototype/let-user-change-answers">Let the user change their answers</a>
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:47]:
```suggestion
              <a href="/how-tos/make-first-prototype/branching">Show different pages depending on user input (branching)</a>
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:50]:
```suggestion
              <a href="/how-tos/make-first-prototype/link-index-page-start-page">Link your index page to your start page</a>
```

File: app/views/how-tos/index.html
Comment (frankieroberto) [inline app/views/how-tos/index.html:67]:
```suggestion
```

File: app/views/how-tos/make-first-prototype/branching.html
Comment (frankieroberto) [inline app/views/how-tos/make-first-prototype/branching.html:11]: Referencing the urls by index number here is clever in reducing duplication, but perhaps kinda hard to read and maintain? eg adding an additional page to the array near the start would break everything?

I’d be tempted to hardcode all these URL references, and then rely on find-and-replace if any pages every get renamed in future?

File: app/views/content-page.html
Comment (frankieroberto) [inline app/views/content-page.html:1]: Delete testing page?

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Looking good!

Would you mind doing indenting on all the nested HTML just to make it easier to read? A code formatter like [prettier](https://www.npmjs.com/package/prettier) should be able to do it for you automatically.

Also the files used to test the tutorial itself (`magical-powers.html` etc) should all be deleted.

Then perhaps we get another pair of eyes on it (@edwardhorsford?), and see if we can find someone to test it by actually following the tutorial?

File: app/views/how-tos/make-first-prototype/branching.html
Comment (vickytnz) [reply app/views/how-tos/make-first-prototype/branching.html:11]: Yeah - this is probably a bit over-designed so will roll back for now, the main thing is that any changes are going to need a lot of checks just as the links are used in several places (similarly as to why I had real pages as a check just to make sure that everything worked).

Comment (vickytnz) [reply app/views/how-tos/make-first-prototype/branching.html]: I've temporarily renamed the elements to things that are self-evident while we continue to test the pages - the observation I had from doing the translation is that since there is a lot of changing pages over multiple screens it's really hard to hardcode everything until it's settled (I found some extra juggling code that I'd missed!)

File: app/views/content-page.html
Comment (vickytnz) [reply app/views/content-page.html:1]: done

**Comment by vickytnz** [comment]
Current status is that this is in a good place for review:
* there may be a few little code things with styling that aren't 100% with the GOV.UK version (whether things are styled as code or strong etc)
* I've removed the example pages but deliberately left all the new content as variables in the base page just to make sure any changes happen across all pages - it could be that they can be hard-coded but it was so tricky to find everything that I don't want to remove them just yet

File: app/config.js
Comment (frankieroberto) [inline app/config.js:5]: Revert back to the serviceName used by the [current website](https://nhsuk-prototype-kit.azurewebsites.net/docs/how-tos) (which this will replace).

```suggestion
  serviceName: 'Prototype kit',
```

**Review by frankieroberto — APPROVED** [review]

**Comment by vickytnz** [comment]
Need to link from the install guide - will make separate pull request.

**Pull request merged**
